dev/com.ibm.ws.transport.http/src/io/openliberty/http/netty/timeout/TimeoutHandler.java

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.http.netty.timeout;
+
+import java.util.concurrent.TimeUnit;
+
+import com.ibm.ws.http.netty.NettyHttpChannelConfig;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.timeout.IdleStateHandler;
+
+// TODO -> Should this be @ChannelHandler.Sharable. Config is all equivalent across the endpoint
+// so might be fine...
+public class TimeoutHandler extends ChannelDuplexHandler{
+
+    // IdleStateHandlers for request reads or persist reads (NETTY)
+    private static final String NETTY_REQUEST_IDLE_HANDLER  = "requestIdleHandler";
+    private static final String NETTY_PERSIST_IDLE_HANDLER  = "persistIdleHandler";
+
+    // TimeoutEventHandlers for request read  or persist user event (OL)
```

Comment:
Not sure I'm following the OL vs Netty here, does OL mean channel framework in this regard or Open Liberty unrelated to channel framework?